src/components/item-list.tsx

```diff
@@ -18,68 +20,88 @@ export const ItemList = ({
+      <hr />
       {Object.keys(itemsNeeded)
-        .filter((key) => itemsNeeded[key as ItemName] > 0)
+        .filter((key) => itemsNeeded[key as ItemName] === 0)
         .sort()
         .map((key) => {
-          if (
-            search.length === 0 ||
-            key.toLocaleLowerCase().includes(search.toLocaleLowerCase())
-          ) {
-            return (
-              <div key={key}>
-                {key}: {itemsNeeded[key as ItemName]}{" "}
-                {quests
-                  .filter(
-                    (quest) => quest.parts.length > questProgress[quest.name]
-                  )
-                  .some((quest) =>
-                    quest.parts.some((part) =>
-                      part.dropItems?.some((item) => item.item === key)
-                    )
-                  )
-                  ? "💀🗑️"
-                  : null}
-              </div>
-            );
+          if (searchCount === 0 || key.toLocaleLowerCase().includes(search.toLocaleLowerCase())) {
+            return (<div key={key}>{key}</div>);
           }
-        })}
-      {Object.keys(itemsNeeded).filter(
-        (key) => itemsNeeded[key as ItemName] > 0
-      ).length === 0
-        ? "No items needed based on current filter criteria."
-        : null}
+        })
+      }
+    </div>);
+
+  const header = (<div>
+    <hr />
+    <div>{deadDrop}️ = Item must be dead dropped</div>
+    {omittedItems !== null ? (<div>
+      <div>{omittedItems.charAt(0).toUpperCase() + omittedItems.slice(1)} items are currently hidden.</div>
+    </div>) : null}
+    <hr />
+  </div>);
+
+  const focusedQuests = (<div>
+    <hr />
+    Only showing items from:
+    <ul className="quest-list">
+      {focusQuests.map((questName) => (
+        <li key={questName}>
+          {questName} - part {questProgress[questName] + 1}
+        </li>
+      ))}
+    </ul>
+  </div>);
+
+  const itemList = (<div>
+    {Object.keys(itemsNeeded)
+      .filter((key) => itemsNeeded[key as ItemName] > 0)
+      .sort()
+      .map((key) => {
+        if (searchCount === 0 || key.toLocaleLowerCase().includes(search.toLocaleLowerCase())) {
+          return (
+            <div key={key}>
+              {key}: {itemsNeeded[key as ItemName]}{" "}
+              {quests
+                .filter((quest) => quest.parts.length > questProgress[quest.name])
+                .some((quest) =>
+                  quest.parts.some((part) =>
+                    part.dropItems?.some((item) => item.item === key)
+                  )
+                )
+                ? deadDrop
+                : null}
+            </div>
+          );
+        }
+      })}
+  </div>);
+
+  return (
+    <div>
+      {searchBox}
+      {itemsCompletedCount === itemsNeededCount ? sellableList : (<div>
+        {header}
+        {focusQuestsCount > 0 ? focusedQuests : null}
+        {itemList}
+        {focusQuestsCount === 0 && omittedItems === null ? sellableList : null}
+      </div>)}
```

Comment:
I can't decide if I like this way of organizing elements. Part of me feels like if you're going to separate all of them, you may as well make them components. Mind convincing me this is the way to go? I agree it's generally more readable than just having all the markup loose, but, what are your thoughts on taking the elements here and making them components versus doing it this way?

Reply:
No argument whatsoever, really. This is my first foray into React; I mostly do backend Go/Python so if anything it was to help myself understand the current structure and learn how to refactor etc. In that spirit, I'm probably in favour of whatever allows maintainability and extensibility. I can see why that might be more granular components, if appropriate.

Reply:
Gotcha. Well I'm impressed, and I was actually going to reach out and ask if you were looking for a job at all, I need to hire a senior FE dev, I love your attention to detail already.

About the components, I know some people will do this kind of thing, I'm not used to it, but that doesn't mean it's wrong. The beauty/ugly of react is you can kinda do whatever and none of it is "wrong". I'm okay with leaving it as-is but if these elements were reused later I think I'd rather see them as components. I might prefer to see them as components anyways just so this file doesn't get too big, as it kind of is, but, having a ton of files with a few lines isn't always helpful either. We can just run with it and switch it up later if it's a problem. I think this is actually a pretty sensible way to organize these elements within a component and I'm just experiencing an existential struggle as I challenge my worldview :)

Reply:
It's always nice to know I'm not the only one learning something through a PR! The apparent structural flexibility of React (JS/TS, really) is quite a shift from what I'm used to, so knowing the pros and cons of a situation is super helpful, thank you. 

If the current state is or gets too unwieldy, refactoring into components sounds like another learning opportunity, so raise an issue if so and I'll give it a spin.

As for the offer, I'm very flattered but in a good spot at the moment I think. You never know what's around the corner though [🤷](https://emojipedia.org/person-shrugging/)